### PR TITLE
Update download link.

### LINKED ISSUE #16414: [v.10.0] /docs/pages/server-access/guides/openssh.mdx
## Details
The [Downloads](https://goteleport.com/docs/docs/cloud/downloads/) link referenced in [Teleport Cloud Prerequisites](https://goteleport.com/docs/server-access/guides/openssh/?scope=cloud#prerequisites) is broken (404).
### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by alexfornuto**
Resolves #16414 by updating the download link to bypass the redirect, which seems to not work in some cases (though I could not replicate).

### CODE REVIEW

**Review by zmb3 — DISMISSED** [review]
This link appears in 5 places. I would either fix the redirect or make this update for all occurrences.

**Comment by alexfornuto** [comment]
> This link appears in 5 places. I would either fix the redirect or make this update for all occurrences.

@zmb3 I updated the link throughout.

File: docs/pages/connect-your-client/gui-clients.mdx [hunk 1]
```diff
@@ -51,7 +51,7 @@ Ensure that your environment includes the following:
   [sign up page](https://goteleport.com/signup/) to begin your free trial.
 
 - The `tsh` client tool version >= (=cloud.version=). To download these tools,
-  visit the [Downloads](/docs/cloud/downloads) page.
+  visit the [Downloads](/docs/deploy-a-cluster/teleport-cloud/downloads/) page.
```
Comment (ptgott) [inline docs/pages/connect-your-client/gui-clients.mdx:54]: We should use relative paths an MDX file for links, e.g., `../deploy-a-cluster/teleport-cloud/downloads.mdx`. That way the docs engine can automatically add the `/ver/<version>` URL path segment for the current version of the docs and any other logic for local docs links.

(Come to think of it, this isn't really written down anywhere, so I can add a linter rule for this.)

Comment (alexfornuto) [reply docs/pages/connect-your-client/gui-clients.mdx:54]: @ptgott how would that work for the link in the include below? I would imagine that the include could be used anywhere, and so the correct relative path would change depending on the number of child directories the doc calling it is in.

Comment (ptgott) [reply docs/pages/connect-your-client/gui-clients.mdx:54]: In includes, relative paths are from the include file. There's logic in the docs engine that resolves the relative paths to relative paths from the main docs page.

Comment (ptgott) [reply docs/pages/connect-your-client/gui-clients.mdx:54]: Added an issue here to flag this in the linter: https://github.com/gravitational/docs/issues/156

**Review by ptgott — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**